crates/astria-sequencer/src/authority/query.rs

```diff
@@ -0,0 +1,273 @@
+        Ok(address_bytes) => address_bytes,
+        Err(_) => {
+            return Err(error_query_response(
+                None,
+                AbciErrorCode::INVALID_PARAMETER,
+                "address was not the correct length",
+            ));
+        }
+    };
+
+    Ok(address_bytes)
+}
+
+fn error_query_response(
+    err: Option<astria_eyre::eyre::Error>,
+    code: AbciErrorCode,
+    msg: &str,
+) -> response::Query {
+    let log = match err {
+        Some(err) => format!("{msg}: {err:#}"),
+        None => msg.into(),
+    };
+    response::Query {
+        code: Code::Err(code.value()),
+        info: code.info(),
+        log,
+        ..response::Query::default()
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::{
+        collections::BTreeMap,
+        vec,
+    };
+
+    use astria_core::protocol::transaction::v1::action::{
+        ValidatorUpdate,
+        ValidatorUpdateV2,
+    };
+    use cnidarium::StateDelta;
+    use tendermint::abci::request;
+
+    use crate::{
+        authority::{
+            query::validator_name_request,
+            StateReadExt,
+            StateWriteExt,
+            ValidatorSet,
+        },
+        test_utils::verification_key,
+    };
+
+    #[tokio::test]
+    async fn validator_name_request_works_as_expected() {
+        let storage = cnidarium::TempStorage::new().await.unwrap();
+        let snapshot = storage.latest_snapshot();
+        let mut state = StateDelta::new(snapshot);
+
+        let verification_key = verification_key(1);
+        let key_address = *verification_key.clone().address_bytes();
+        let validator_name = "test".to_string();
+        let inner_update = ValidatorUpdate {
+            power: 100,
+            verification_key: verification_key.clone(),
+        };
+        let update_with_name = ValidatorUpdateV2 {
+            name: validator_name.clone(),
+            power: 100,
+            verification_key,
+        };
+
+        let mut validator_names = state.get_validator_names().await.unwrap();
+        assert_eq!(validator_names.len(), 0);
+
+        let inner_validator_map = BTreeMap::new();
+        let mut validator_set = ValidatorSet::new(inner_validator_map);
+        assert_eq!(validator_set.len(), 0);
+
+        validator_names.push_name(&key_address, update_with_name.name.clone());
+        validator_set.push_update(inner_update);
+
+        state.put_validator_names(validator_names).unwrap();
+        state.put_validator_set(validator_set).unwrap();
+        storage.commit(state).await.unwrap();
+
+        let query = request::Query {
+            data: vec![].into(),
+            path: "path".to_string(),
+            height: 0u32.into(),
+            prove: false,
+        };
+        let params = vec![("address".to_string(), hex::encode(key_address))];
+
+        let rsp = validator_name_request(storage.clone(), query, params).await;
+        assert_eq!(rsp.code, 0.into(), "{}", rsp.log);
```

Comment:
The goal here was to provide some more information in the error message. For example, if I cause the test to err, right now it will output this:

``` 
assertion `left == right` failed: address could not be constructed from provided parameter: failed to parse argument as address: failed decoding provided string: invalid checksum: the checksum residue is not valid for the data
  left: Err(2)
 right: Ok
```

but if just using `assert`, it will output this:

```
address could not be constructed from provided parameter: failed to parse argument as address: failed decoding provided string: invalid checksum: the checksum residue is not valid for the data
```

I suppose the log is probably plenty detailed enough, but had just figured it may be nice to also be given back the error code itself if trying to debug.

Reply:
We can always add the code to the error message in the `assert!` like
```rust
assert!(rsp.code.is_ok(), "code: {:?}, log: {}", rsp.code, rsp.log);
```
to yield output like
```
code: Err(2), log: address could not be constructed from provided parameter: failed to parse argument as address: failed decoding provided string: invalid checksum: the checksum residue is not valid for the data
```
But I'm not too bothered about this - happy for you to leave as is :)

Reply:
I actually like format of your suggestion a lot better. Made that change in 29496c7.